.github/workflows/sketcher-builder.yml

```diff
@@ -0,0 +1,59 @@
+name: Sketcher Build and Test
+
+on:
+  workflow_dispatch: ~
+  push: ~
+
+jobs:
+  # TODO: clang-format hook
+  build_test_sketcher:
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "Ubuntu",
+            os: ubuntu-latest,
+            app: "build/sketcher_app",
+          }
+        - {
+            name: "macOS",
+            os: macos-latest,
+            app: "build/sketcher_app",
+          }
+        - {
+            name: "Windows",
+            os: windows-latest,
+            # TODO: why does windows drop this in Debug?
+            app: 'build/Debug/sketcher_app.exe',
+          }
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      # TODO: store/read versions in cmake-accessible way
+      - name: Dependency cache - fmt-10.1.0
+        uses: actions/cache@v4
+        with:
+          path: build/_deps/fmt-*
+          key: fmt-10.1.0-${{ matrix.config.os }}
+
+      - name: Dependency cache - zstd-1.5.5
+        uses: actions/cache@v4
+        with:
+          path: build/_deps/zstd-*
+          key: zstd-1.5.5-${{ matrix.config.os }}
+
+      - name: Build sketcher
+        run: |
+          cmake . -B build -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release
```

Comment:
Since you are adding the option to the CMakefile (yay, more stuff to maintain), you should also provide a CI to build with shared libs.

Reply:
Ah, you're right. I'm gonna punt on getting this working for SHARED for the time being and soldier on with setting up static building.

Reply:
Oh, unless you think I should just lock this all down on buiding static only, I'd happily do that.

Reply:
I think the sketcher makes more sense for users as a static executable than an executable plus a set of shared libraries (easier to handle, harder to have dependency issues). Not sure if you have something else in mind.

Reply:
Cool, switched the CMake over to always build a static library exclusively.